torch/csrc/api/src/nn/modules/batchnorm.cpp

```diff
@@ -3,15 +3,13 @@
 namespace torch { namespace nn {
 void BatchNorm::initialize_parameters() {
   if (affine_) {
-    weight = this->add(
-        Var(DefaultTensor(at::kFloat).tensor(num_features_), true), "weight");
-    bias = this->add(
-        Var(DefaultTensor(at::kFloat).tensor(num_features_), true), "bias");
+    weight = this->add(Var(at::CPU(at::kFloat).empty(num_features_)), "weight");
+    bias = this->add(Var(at::CPU(at::kFloat).empty(num_features_)), "bias");
   }
 
   if (stateful_) {
-    running_mean = Var(DefaultTensor(at::kFloat).zeros({num_features_}), false);
-    running_var = Var(DefaultTensor(at::kFloat).ones({num_features_}), false);
+    running_mean = Var(at::CPU(at::kFloat).zeros({num_features_}), false);
```

Comment:
Why are these hard coded CPU now?

Reply:
Well I think it's usual (e.g. in Python PyTorch) that all tensors always start on the CPU, and one then calls `.cuda()` to move them over?